File: Solutions/2024/Scala/src/test/scala/Day2Tests.scala

import Day2.*

class Day2Tests extends munit.FunSuite:

  test("Part1 TestInput") {
    val input = Day2.parseFile("/Day2/TestInput1.txt")
    val obtained = Part1(input)
    val expected = 2
    assertEquals(obtained, expected)
  }

  test("Part1") {
    val input = Day2.parseFile("/Day2/Input.txt")
    val obtained = Part1(input)
    val expected = 670
    assertEquals(obtained, expected)
  }

  test("Part2 TestInput") {
    val input = Day2.parseFile("/Day2/TestInput1.txt")
    val obtained = Part2(input)
    val expected = 4
    assertEquals(obtained, expected)
  }

  test("Part2") {
    val input = Day2.parseFile("/Day2/Input.txt")
    val obtained = Part2(input)
    val expected = 700
    assertEquals(obtained, expected)
  }

File: Solutions/2025/Java/Java/src/test/java/Day2Tests.java
import Day2.Day2;
import org.junit.jupiter.api.Test;
import static org.junit.Assert.assertEquals;

class DayXTests {

    @Test
    void Part1TestInput() {
        var day = new Day2();
        var input = day.ParseInput("Day2/TestInput.txt");
        var result = day.Part1(input);
        assertEquals(1227775554L, result);
    }

    @Test
    void Part1() {
        var day = new Day2();
        var input = day.ParseInput("Day2/Input.txt");
        var result = day.Part1(input);
        assertEquals(32976912643L, result);
    }

    @Test
    void Part2TestInput() {
        var day = new Day2();
        var input = day.ParseInput("Day2/TestInput.txt");
        var result = day.Part2(input);
        assertEquals(4174379265L, result);
    }

    @Test
    void Part2() {
        var day = new Day2();
        var input = day.ParseInput("Day2/Input.txt");
        var result = day.Part2(input);
        assertEquals(54446379122L, result);
    }
}